tests/gold_tests/autest-site/microserver.test.ext

```diff
@@ -160,16 +180,27 @@ def MakeOriginServer(obj, name, port=False, ip=False, delay=False, ssl=False, lo
     for flag, value in options.items():
         command += " {} {}".format(flag, value)
 
-    # create process
     p.Command = command
     p.Setup.MakeDir(data_dir)
     p.Variables.DataDir = data_dir
     p.Variables.lookup_key = lookup_key
-    p.Ready = When.PortOpen(port, ip)
-    p.ReturnCode = Any(None, 0)
     AddMethodToInstance(p, addResponse)
     AddMethodToInstance(p, addTransactionToSession)
 
+    # Set up health check.
+    addResponse(p, "sessionfile.log", {
+        "headers": "GET /ruok HTTP/1.1\r\nHost: {}\r\n\r\n".format(ip),
+        "timestamp": "1469733493.993",
+        "body": ""
+    }, {
+        "headers": "HTTP/1.1 200 OK\r\nConnection: close\r\n\r\n",
+        "timestamp": "1469733493.993",
+        "body": "imok"
+    })
+
+    p.Ready = When.uServerUpAndRunning(ip, port, ssl)
```

Comment:
Do we have any case when this may return false?

Reply:
The `p.Ready =` statement stores the method pointed to for future execution, so the rvalue is just the method.

The `uServerUpAndRunning`, when executed, should either return `True` (check passed), raise an exception (unexpected response), or just keep trying with a sleep interval of `.1` seconds. As I understand it, there is a timeout that is watched somewhere in the caller chain that will "fail" (error) the test if reached before the check returns.

Reply:
Actually, when I tested this assumption, I found that this seems to check for a timeout in the same thread, so I need to return `False` rather than loop on an interval. If something strange prevents the server from listening for TCP connections, then we would never error out. I'll fix that.